manila/share/manager.py

```diff
@@ -4018,21 +4021,18 @@ def delete_snapshot(self, context, snapshot_id, force=False,
                     resource_id=snapshot_instance_id,
                     exception=excep)
 
-        self.db.share_snapshot_instance_delete(context, snapshot_instance_id)
-        self._delete_snapshot_quota(context, snapshot_ref)
-
-    def _get_snapshot_instances_with_deletion_deferred(self, ctxt):
-        snap_instances = self.db.share_snapshot_instance_get_all_with_filters(
-            ctxt, {'statuses': constants.STATUS_DEFERRED_DELETING})
+        snapshot_instance = self.db.share_snapshot_instance_get(
+            context, snapshot_ref.instance['id'])
+        if snapshot_instance['status'] == (
+                constants.STATUS_ERROR_DEFERRED_DELETING) and not force:
+            return
 
-        snap_instances_error_deferred_deleting = \
-            self.db.share_snapshot_instance_get_all_with_filters(
-                ctxt, {'statuses': constants.STATUS_ERROR_DEFERRED_DELETING})
-        updated_del = timeutils.utcnow() - datetime.timedelta(minutes=30)
-        for snap_instance in snap_instances_error_deferred_deleting:
-            if snap_instance.get('updated_at') < updated_del:
-                snap_instances.append(snap_instance)
-        return snap_instances
+        self.db.share_snapshot_instance_delete(context, snapshot_instance_id)
+        if snapshot_instance['status'] not in (
+            constants.STATUS_DEFERRED_DELETING,
+            constants.STATUS_ERROR_DEFERRED_DELETING
+        ):
```

Comment:
change if condition to  `if deferred_delete is False`

Reply:
correct. But current if check is optimised. e.g.  
if we are deferred deleting means snapshot is in STATUS_DEFERRED_DELETING.
if we are forcefully deleting snapshot which is in STATUS_ERROR_DEFERRED_DELETING, means quota is already cleaned. So just checking presence of either states means we dont need to clean quota.